ariadne/cardano/src/Ariadne/System/FileMode.hs

```diff
@@ -0,0 +1,39 @@
+module Ariadne.System.FileMode
+       ( getAccessMode
+       , setMode
+       , ensureModeIs
+       , mode600
+       ) where
+
+import System.Wlog (logWarning, usingLoggerName, CanLog)
+
+import qualified System.Posix.Files as PSX
+
+import qualified System.Posix.Types as PSX (FileMode)
+
+-- | It's common practice to make keyfiles readable/writable for user only.
+-- For now keyfile is read-only
+mode600 :: PSX.FileMode
+mode600 = PSX.unionFileModes PSX.ownerReadMode PSX.ownerWriteMode
+
+-- | Return only the access part of the file mode (like owner:rw-, etc).
+getAccessMode :: MonadIO m => FilePath -> m PSX.FileMode
+getAccessMode path = do
+    mode <- liftIO $ PSX.fileMode <$> PSX.getFileStatus path
+    return $ PSX.intersectFileModes mode PSX.accessModes
+
+-- | Set mode disregard current mode of the file.
+setMode :: MonadIO m => PSX.FileMode -> FilePath -> m ()
+setMode mode path = liftIO $ PSX.setFileMode path mode
+
+-- | Set given mode if needed.
+ensureModeIs
+    :: (MonadIO m, CanLog m)
```

Comment:
Do I understand correctly that this function is missing the `IsPosix` constraint?

Reply:
I thought so to, but adding it there triggers `warning: [-Wredundant-constraints]`

Reply:
Please see the discussion in #haskell in Slack. The whole approach with `IsPosix` is not going to work. I suggest that you get rid of all mentions of it.

Reply:
IIUC In that discussion in #haskell you tried to use `IsPosix` constraint to make sure that `whenPosix` _is not_ called when it _is not_ necessary. while its original purpose was to ensure `whenPosix` _is_ used when it _is_ necessary.